### PR TITLE
Add a regression test for recent regression in pair.

### PR DESCRIPTION
**Pull request by jyknight**
PR #96165 broke code similar to this test, and was subsequently reverted. Add a test-case, to ensure the problem won't reoccur.

This error is potentially related to issues #59292 and #59966.

### CODE REVIEW

**Comment by frederick-vs-ja** [comment]
Seems related to #59292 and #59966. Should we mention these issues?

File: libcxx/test/libcxx/utilities/utility/pairs/pairs.pair/pair.incomplete.verify.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Check that instantiating pair doesn't look up type traits "too early", before
```
Comment (philnik777) [inline libcxx/test/libcxx/utilities/utility/pairs/pairs.pair/pair.incomplete.verify.cpp:1]: Please add a license header

**Review by ldionne — APPROVED** [review]
LGTM w/ comments addressed. Thanks!

File: libcxx/test/libcxx/utilities/utility/pairs/pairs.pair/pair.incomplete.verify.cpp (hunk 1 above)
Comment (jyknight) [reply libcxx/test/libcxx/utilities/utility/pairs/pairs.pair/pair.incomplete.verify.cpp:1]: Done.

File: libcxx/test/libcxx/utilities/utility/pairs/pairs.pair/pair.incomplete.verify.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// the contained types have been completed.
+//
+// This is a regression test, to prevent a reoccurrance of the issue introduced in
+// 5e1de27f680591a870d78e9952b23f76aed7f456.
+
+// expected-no-diagnostics
```
[the remark this answers is not in the document]
Comment (jyknight) [reply libcxx/test/libcxx/utilities/utility/pairs/pairs.pair/pair.incomplete.verify.cpp:7]: Done.

**Review by jyknight — COMMENTED** [review]
Added a reference to the possibly-related issues as suggested.

**Review by philnik777 — APPROVED** [review]

**Pull request merged**